dataflow/gpu-workers/e2e_test.py

```diff
@@ -0,0 +1,190 @@
+    )
+    assert python_version == "Python 3.6.9"
+
+
+def test_apache_beam_version(image_name: str, configure_docker: None) -> None:
+    # Make sure the installed Apache Beam version matches the Apache Beam image
+    # we use to copy the worker boot file.
+    # If this test fails, the following needs updating:
+    # - Dockerfile: The `COPY --from=apache/beam` for the worker boot file.
+    apache_beam_version = (
+        subprocess.run(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-i",
+                "--entrypoint=bash",
+                image_name,
+                "-c",
+                "pip freeze | egrep '^apache-beam=='",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.decode("utf-8")
+        .strip()
+    )
+    assert apache_beam_version == "apache-beam==2.26.0"
+
+
+def test_tensorflow_version(image_name: str, configure_docker: None) -> None:
+    # Make sure the installed Tensorflow version matches the Tensorflow version
+    # in the Dockerfile.
+    # If this test fails, the following needs updating:
+    # - Dockerfile: The `FROM tensorflow/tensorflow` version.
+    tensorflow_version = (
+        subprocess.run(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-i",
+                "--entrypoint=bash",
+                image_name,
+                "-c",
+                "pip freeze | egrep '^tensorflow(-gpu)?=='",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.decode("utf-8")
+        .strip()
+    )
+    assert tensorflow_version == "tensorflow-gpu==2.4.0"
+
+
+def test_end_to_end(bucket_name: str, image_name: str) -> None:
+    # Run the Beam pipeline in Dataflow making sure GPUs are used.
+    gpu_type = "nvidia-tesla-t4"
+    region = "us-central1"
+    worker_zone = "us-central1-a"
+    subprocess.run(
+        [
+            "python",
+            "landsat_view.py",
+            f"--output-path-prefix=gs://{bucket_name}/outputs/",
+            "--runner=DataflowRunner",
+            f"--project={PROJECT}",
+            f"--region={region}",
+            "--worker_machine_type=custom-1-13312-ext",
+            f"--worker_harness_container_image={image_name}",
+            f"--worker_zone={worker_zone}",
+            f"--experiments=worker_accelerator=type={gpu_type},count=1,install-nvidia-driver",
+            "--experiments=use_runner_v2",
+        ],
+        check=True,
+    )
+
+    # Check that output files were created and are not empty.
+    storage_client = storage.Client()
+    output_files = list(storage_client.list_blobs(
+        bucket_name, prefix="outputs/"))
+    assert len(output_files) > 0, "No output files found"
+    for output_file in output_files:
+        assert output_file.size > 0, f"Output file is empty: {output_file.name}"
+        output_file.delete()
```

Comment:
If you add the `force=True` to your bucket delete, you don't need to worry about deleting the files here as cleanup. 

If however you do want to go through the bucket and delete the files, you should probably do so in the cleanup part of the fixture rather than in the test itself, just in case something goes wrong in the test

Reply:
Thanks! I didn't know that was an option!